middleware/kubernetes/kubernetes.go

```diff
@@ -190,11 +190,27 @@ func (k *Kubernetes) getZoneForName(name string) (string, []string) {
 // stripSRVPrefix separates out the port and protocol segments, if present
 // If not present, assume all ports/protocols (e.g. wildcard)
 func stripSRVPrefix(name []string) (string, string, []string) {
-	if name[0][0] == '_' && name[1][0] == '_' {
-		return name[0][1:], name[1][1:], name[2:]
+
+	var srv, prt string
+
+	if len(name) != 5 {
```

Comment:
yes - name is perhaps a bad (or at least ambiguous) choice of variable name...  it does not contain the zone.  A length of 5 occurs when there is a service and protocol present.  In that case, it comprises _service._protocol.service.namespace.svc.